File: server/datasets/java_source-code/Multiple-B2/organic_9.java
// Two Sum (hash map) - organic submission 9
// Pair search scanning for the first valid complement.

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (seen.containsKey(complement)) {
                return new int[]{seen.get(complement), i};
            }
            seen.put(nums[i], i);
        }
        return null;
    }

    public static void main(String[] args) {
        int[] nums = {0, 4, 3, 0};
        int target = 0;
        System.out.println("Indices: " + Arrays.toString(twoSum(nums, target)));
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_b2/organic_9.py
# Two Sum (hash map) - organic submission 9
# Generator-based: yields every valid pair, caller takes the first one.

def two_sum_pairs(nums, target):
    seen = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in seen:
            yield [seen[complement], i]
        seen[num] = i


def two_sum(nums, target):
    for pair in two_sum_pairs(nums, target):
        return pair
    return None


if __name__ == "__main__":
    nums = [0, 4, 3, 0]
    target = 0
    print("Indices:", two_sum(nums, target))
